platform/admin/core/admin-core-insecuredefaults/src/main/java/org/codice/ddf/admin/insecure/defaults/service/UserPropertiesDeletionScheduler.java

```diff
@@ -0,0 +1,150 @@
+import org.apache.camel.impl.DefaultCamelContext;
+import org.codice.ddf.configuration.AbsolutePathResolver;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class UserPropertiesDeletionScheduler {
+
+  static final Path TEMP_FILE_TIMESTAMP_PATH =
+      Paths.get(new AbsolutePathResolver("data/tmp/timestamp.txt").getPath());
+  static final Path USERS_PROPERTIES_FILE_PATH =
+      Paths.get(new AbsolutePathResolver("etc/users.properties").getPath());
+  private static final Logger LOGGER =
+      LoggerFactory.getLogger(UserPropertiesDeletionScheduler.class);
+  private static boolean deletionScheduled;
+
+  public UserPropertiesDeletionScheduler() {
+    deletionScheduled = false;
+  }
+
+  public void scheduleDeletion(Path path) {
+    if (!deletionScheduled) {
+      String cron = getCron(path);
+      if (cron == null) {
+        return;
+      }
+      try {
+        CamelContext context = new DefaultCamelContext();
+        RouteBuilder builder =
+            new RouteBuilder() {
+              @Override
+              public void configure() throws Exception {
+                fromF("quartz2://scheduler/deletionTimer?cron=%s", cron)
+                    .bean(UserPropertiesDeletionScheduler.class, "deleteFiles");
+              }
+            };
+
+        context.addRoutes(builder);
+        context.start();
+        deletionScheduled = true;
+      } catch (Exception e) {
+        LOGGER.debug("Unable to create route", e);
+      }
+    }
+  }
+
+  @VisibleForTesting
+  String getCron(Path path) {
+    try {
+      if (!path.toFile().exists()) {
+        deleteFiles();
+        return null;
+      }
+
+      if (!TEMP_FILE_TIMESTAMP_PATH.toFile().exists()) {
+
+        File tempFile = new File(TEMP_FILE_TIMESTAMP_PATH.toString());
+        if (!tempFile.createNewFile()) {
+          return null;
+        }
+        Files.write(
+            TEMP_FILE_TIMESTAMP_PATH,
+            Long.toString(System.currentTimeMillis()).getBytes(StandardCharsets.UTF_8));
+
+        return cronCalculator(new Date(System.currentTimeMillis()));
+
+      } else {
+        Date firstInstall = installationDate();
+        if (firstInstall == null) {
+          return null;
+        }
+        Date timestamp = new Date(firstInstall.getTime() + TimeUnit.DAYS.toMillis(3));
+        Date currentDate = new Date();
+        if (currentDate.before(timestamp)) {
+          return cronCalculator(firstInstall);
+        }
+        deleteFiles();
+      }
+    } catch (IOException e) {
+      LOGGER.debug("Unable to access the temporary file", e);
+    }
+    return null;
+  }
+
+  public void deleteFiles() {
+    try {
+      Files.deleteIfExists(USERS_PROPERTIES_FILE_PATH);
+      Files.deleteIfExists(TEMP_FILE_TIMESTAMP_PATH);
+      deletionScheduled = false;
+    } catch (IOException e) {
+      LOGGER.debug("Unable to delete file");
+    }
+  }
+
+  @VisibleForTesting
+  String cronCalculator(Date firstInstall) {
```

Comment:
✏️ Using an `Instant` here, we could avoid converting to a `Calendar` object for extraction of field data.